### PR TITLE
improve mutation log: remove useless variable; use atomic_bool instea…

### PR DESCRIPTION
**Pull request by qinzuoyan**
…d of volatile bool

### CODE REVIEW

File: src/dist/replication/lib/mutation_log.cpp [hunk 1]
```diff
@@ -190,7 +189,7 @@ void mutation_log_shared::write_pending_mutations(bool release_lock)
             // here we use _is_writing instead of _issued_write.expired() to check writing done,
             // because the following callbacks may run before "block" released, which may cause
             // the next init_prepare() not starting the write.
-            _is_writing = false;
+            _is_writing.store(false, std::memory_order_release);
```
Comment (imzhenyu) [inline src/dist/replication/lib/mutation_log.cpp]: If I understand correctly, we should us acquire in this case as we don't want to the latter callback ops to be scheduled first. release seems unnecessary for me.

File: src/dist/replication/lib/mutation_log.cpp [hunk 2]
```diff
@@ -165,7 +164,7 @@ void mutation_log_shared::write_pending_mutations(bool release_lock)
         mus = std::move(pmu)
         ](error_code err, size_t sz) mutable
         {
-            dassert(_is_writing, "");
+            dassert(_is_writing.load(std::memory_order_acquire), "");
```
Comment (imzhenyu) [inline src/dist/replication/lib/mutation_log.cpp]: relax?

File: src/dist/replication/lib/mutation_log.cpp [hunk 3]
```diff
@@ -418,7 +416,7 @@ void mutation_log_private::write_pending_mutations(bool release_lock)
             // here we use _is_writing instead of _issued_write.expired() to check writing done,
             // because the following callbacks may run before "block" released, which may cause
             // the next init_prepare() not starting the write.
-            _is_writing = false;
+            _is_writing.store(false, std::memory_order_release);
```
Comment (imzhenyu) [inline src/dist/replication/lib/mutation_log.cpp]: same as above

File: src/dist/replication/lib/mutation_log.cpp (hunk 1 above)
Comment (shengofsun) [inline src/dist/replication/lib/mutation_log.cpp]: @imzhenyu  Here could be relaxed, I think it doesn't matter if callbacks schedules first.

File: src/dist/replication/lib/mutation_log.cpp (hunk 2 above)
Comment (shengofsun) [inline src/dist/replication/lib/mutation_log.cpp]: Here should be relaxed indeed.

**Comment by qinzuoyan** [comment]
@imzhenyu @shengofsun , fixed according to your comments.

File: src/dist/replication/lib/mutation_log.cpp (hunk 1 above)
Comment (imzhenyu) [inline src/dist/replication/lib/mutation_log.cpp]: @qinzuoyan @shengofsun I think you guys are correct - just found it crash on Windows. I'm falling it back to memory_order_relaxed.

**Pull request merged**